gpflow/experimental/check_shapes/specs.py

```diff
@@ -24,30 +24,36 @@
 class ParsedDimensionSpec:
     constant: Optional[int]
     variable_name: Optional[str]
+    variable_rank: bool
 
     def __post_init__(self) -> None:
         assert (self.constant is None) != (
             self.variable_name is None
         ), "Argument must be either constant or variable."
+        if self.variable_rank:
+            assert self.constant is None, "Constants cannot have a variable rank."
 
     def __repr__(self) -> str:
         if self.constant is not None:
             return str(self.constant)
         else:
             assert self.variable_name is not None
-            return self.variable_name
+            suffix = "..." if self.variable_rank else ""
+            return self.variable_name + suffix
 
 
 @dataclass(frozen=True)
 class ParsedShapeSpec:
-    leading_dims_variable_name: Optional[str]
-    trailing_dims: Tuple[ParsedDimensionSpec, ...]
+    dims: Tuple[ParsedDimensionSpec, ...]
+
+    def __post_init__(self) -> None:
+        n_variable_rank = sum(dim.variable_rank for dim in self.dims)
+        assert (
+            n_variable_rank <= 1
+        ), "At most one variable-rank dimension allow. Found {n_variable_rank} in {self}."
```

Comment:
Format string missing?
```suggestion
        ), f"At most one variable-rank dimension allow. Found {n_variable_rank} in {self}."
```

Reply:
Of course. Good catch! Thanks!